### PR TITLE
new tutorial structure

### PR DESCRIPTION
**Pull request by SchniddeMitBrot**
- Rearranged all the necessary tutorials to the new structure.
- adjusted the getting-started-spring-boot-maven tutorial to the jQAssistant version 2.0.3

### CODE REVIEW

File: readme.adoc [hunk 1]
```diff
@@ -16,3 +16,147 @@ There are several ways to help and contribute:
 - If you miss a topic, please link:./issues/^[create a issue^] for
   the topic we should cover
 
+= Markup Test for jQAssistant 101
```
Comment (StephanPirnbaum) [inline readme.adoc]: We should not forget to remove this later :D

File: getting-started-spring-boot-maven/jqassistant/index.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!--1-->
+<jqassistant-rules
+        xmlns="http://schema.jqassistant.org/rule/v2.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://schema.jqassistant.org/rule/v2.0 https://schema.jqassistant.org/rule/jqassistant-rule-v2.0.xsd">
+
+    <!--2-->
+    <group id="Default">
+        <includeGroup refId="*:Default"/>
```
Comment (StephanPirnbaum) [inline getting-started-spring-boot-maven/jqassistant/index.xml]: I'm more a fan of explicitly including the groups. Please change to include maven:Default and package:Default

File: getting-started-spring-boot-maven/jqassistant/maven.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<jqassistant-rules
+        xmlns="http://schema.jqassistant.org/rule/v2.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://schema.jqassistant.org/rule/v2.0 https://schema.jqassistant.org/rule/jqassistant-rule-v2.0.xsd">
+
+    <!-- tag::content[] -->
+    <!--1-->
+    <group id="maven:Default">
+        <includeConcept refId="maven:*"/>
```
Comment (StephanPirnbaum) [inline getting-started-spring-boot-maven/jqassistant/maven.xml]: Same goes here. Please change to explicitly stating the concept name

File: getting-started-spring-boot-maven/jqassistant/package.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<jqassistant-rules
+        xmlns="http://schema.jqassistant.org/rule/v2.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://schema.jqassistant.org/rule/v2.0 https://schema.jqassistant.org/rule/jqassistant-rule-v2.0.xsd">
+
+    <!-- tag::content[] -->
+    <!--1-->
+    <group id="package:Default">
+        <includeConstraint refId="package:*"/>
```
Comment (StephanPirnbaum) [inline getting-started-spring-boot-maven/jqassistant/package.xml]: Please explicitly state constraint name

File: getting-started-spring-boot-maven/pom.xml [hunk 5]
```diff
@@ -56,28 +56,13 @@
             <plugin>
                 <groupId>com.buschmais.jqassistant</groupId>
                 <artifactId>jqassistant-maven-plugin</artifactId>
-                <version>1.10.1</version>
+                <version>2.0.3</version>
```
Comment (StephanPirnbaum) [inline getting-started-spring-boot-maven/pom.xml]: Can now be 2.0.4

File: getting-started-spring-boot-maven/readme.adoc [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+[source,xml]
+.pom.xml
+----
+include::pom.xml[tag=plugin,indent=0]
+----
+
+The configuration above...
+
+- integrates the `jqassistant-maven-plugin` with version `2.0.3` into your Maven project.
```
Comment (StephanPirnbaum) [inline getting-started-spring-boot-maven/readme.adoc]: Now 2.0.4

File: getting-started-spring-boot-maven/readme.adoc [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+WARNING: Via the https://github.com/jqassistant-plugin/jqassistant-asciidoc-report-plugin[Asciidoc-report-plugin] it is also possible to use Asciidoc for the rule definition. But this practice is not recommended anymore!
+
+=== Defining your own rules
+
+In this tutorial the xml files are used to create a custom Default group with underlying concepts and constraints.
+Therefore, the directory contains the files `index.xml`, `package.xml` and `maven.xml`.
+
+
+.jqassistant/index.xml
```
Comment (StephanPirnbaum) [inline getting-started-spring-boot-maven/readme.adoc]: I'd vote for naming this someting like "project-default.xml"

**Review by StephanPirnbaum — CHANGES_REQUESTED** [review]

File: getting-started-spring-boot-maven/.jqassistant.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+jqassistant:
+  plugins:
+    #<1>
+    - group-id: org.jqassistant.plugin
+      artifact-id: jqassistant-spring-plugin
+      version: 2.0.0
+    #<2>
+    - group-id: org.jqassistant.plugin
+      artifact-id: jqassistant-plantuml-report-plugin
+      version: 2.0.0
```
Comment (DirkMahler) [inline getting-started-spring-boot-maven/.jqassistant.yml]: please use version 2.0.1

File: getting-started-spring-boot-maven/jqassistant/index.xml (hunk 2 above)
Comment (DirkMahler) [reply getting-started-spring-boot-maven/jqassistant/index.xml]: It depends: agreeing with @StephanPirnbaum , please change to concrete groups here.

File: getting-started-spring-boot-maven/jqassistant/maven.xml (hunk 3 above)
Comment (DirkMahler) [reply getting-started-spring-boot-maven/jqassistant/maven.xml]: Usually I do something like this:

    <includeConstraint refId="maven:*" />
    <includeConcept refId="maven:*Report" />

The idea behind: make sure that all newly added constraints and report concepts are included without adding unnecessary concepts.

File: getting-started-spring-boot-maven/jqassistant/maven.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    </group>
+
+    <!--2-->
+    <concept id="maven:ProjectArtifactDependencies">
```
Comment (DirkMahler) [inline getting-started-spring-boot-maven/jqassistant/maven.xml]: This is a concept just for reporting purposes, it should have an according suffix in its id, e.g. "maven:ProjectArtifactDependencyReport"

File: getting-started-spring-boot-maven/readme.adoc [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+.jqassistant/package.xml
+....
+include::jqassistant/package.xml[tag=content, indent=1]
+....
+<1> This defines a matching group `package:Default` (matches the `\*:Default` pattern from the `Default` group) which itself includes all constraints matching `package:*`
+<2> The example constraint `package:ServicePackage` ensures that Spring services (i.e. classes annotated with `@Service`) are located in packages named `service`.
+
+[[PlantumlDiagramUse]]
+.jqassistant/maven.adoc
```
Comment (DirkMahler) [inline getting-started-spring-boot-maven/readme.adoc]: ```.jqassistant/maven.xml```

**Review by DirkMahler — CHANGES_REQUESTED** [review]

**Review by StephanPirnbaum — APPROVED** [review]

**Review by DirkMahler — APPROVED** [review]

**Pull request merged**